### PR TITLE
Access Rules Post

### PR DESCRIPTION
**Pull request by sijumoncy**
- Post Api implementation for Access Rules
- #457

### CODE REVIEW

**Comment by kavitharaju** [comment]
@sijumoncy Could you please remind me, did we make a decision about gql_tests? I see them failing...

File: app/crud/auth_crud.py [hunk 1]
```diff
@@ -100,3 +101,34 @@ def update_role(db_: Session, role_details, user_id=None):
...
+        db_role = db_.query(db_models.Roles).filter(
+            func.lower(db_models.Roles.roleName) == func.lower(role.strip())).first()
+        if not db_role:
+            raise NotAvailableException(f"role, {role.strip()},"+\
+                " not found in Database")
+
+    db_content = db_models.AccessRules(
+        entitlementId=entitlement.resourceTypeId,
+        tagId=tag.permissionId,
+        roles=details.roles,
```
Comment (kavitharaju) [inline app/crud/auth_crud.py]: Is roles a multi-valued field? Did we make that so in the DB schema? Shouldn't we make them multiple rows, I wonder?

File: app/routers/auth_api.py [hunk 2]
```diff
@@ -375,7 +375,29 @@ async def update_roles(role_details:schema_auth.RoleIn,request: Request,#pylint:
...
+user_details =Depends(get_user_or_none),
+db_: Session = Depends(get_db)):#pylint: disable=unused-argument
+    '''Create Authentication Access Rules
+    * entitlement (ResourceType) is mandatory
+    * tag (permission) is mandatory
+    * roles (List of role) is mandatory'''
+    log.info('In create Access Rules')
+    log.debug('Access rule  Create In:%s',details)
+    data = create_access_rules(db_, details, user_id = user_details['user_id'])
+    print("returned data ------------->", data)
```
Comment (kavitharaju) [inline app/routers/auth_api.py]: could avoid the prints.

**Comment by sijumoncy** [comment]
> @sijumoncy Could you please remind me, did we make a decision about gql_tests? I see them failing...

Yes , We decided to avoid those temporally, using the command

**Comment by kavitharaju** [comment]
Okay then please change the test running command in the CI/CD.

File: app/crud/auth_crud.py (hunk 1 above)
Comment (sijumoncy) [reply app/crud/auth_crud.py]: We are keeping the roles as an array for a pair of entitlement and tagId. 
If needed we can change it to multiple line  like 
E1, T1, R1 - E1 , T1, R2 (where E1, T1, R1 will be a constrain unique )

Comment (kavitharaju) [reply app/crud/auth_crud.py]: Yes, that is how we usually do it in DBs.

**Review by kavitharaju — APPROVED** [review]

**Comment by kavitharaju** [comment]
@sijumoncy I see the attempt to use --ignore-glob in the pytest command hasn't worked. Another one you can try is this
`pytest -k "not test_gql"`.

But merging this PR for now.

**Pull request merged**